src/classes/PMT_Payment_TDTM.cls

```diff
@@ -286,16 +286,41 @@ public class PMT_Payment_TDTM extends TDTM_Runnable {
     * @return void Does not return anything
     ********************************************************************************************************/
     private void validatePayments(
-        List<npe01__OppPayment__c> newlist, 
+        List<npe01__OppPayment__c> newlist,
         List<npe01__OppPayment__c> oldlist,
-        TDTM_Runnable.Action triggerAction, 
+        TDTM_Runnable.Action triggerAction,
         DmlWrapper dmlWrapper) {
         if (triggerAction == TDTM_Runnable.Action.AfterInsert || triggerAction == TDTM_Runnable.Action.AfterUpdate) {
             for (npe01__OppPayment__c payment : newlist) {
+                enforceAccountingDataConsistency(payment);
+
                 if (payment.npe01__Paid__c == true && payment.npe01__Written_Off__c == true) {
                     payment.addError(System.Label.pmtWriteOffPaidPayment);
                 }
             }
         }
     }
+
+    /*******************************************************************************************************
+    * @description If the custom setting to enforce account rules is enabled, it will validate that paid payments
+    * have a payment date and unpaid payments have a scheduled date.
+    * @param payment the Payment from trigger new.
+    * @return void Does not return anything
+    ********************************************************************************************************/
+    private void enforceAccountingDataConsistency(npe01__OppPayment__c payment) {
+        Boolean enforceAccountingDataConsistencyCustomSetting = UTIL_CustomSettingsFacade.getContactsSettings().Enforce_Accounting_Data_Consistency__c;
```

Comment:
Rather than call the `UTIL_CustomSettingsFacade.getContactsSettings()` method in this method (which gets called for each Payment record), I recommend retrieving the Contacts settings earlier in the class and then checking for `Enforce_Accounting_Data_Consistency__c == true` in the `for` loop above so that this method is only called when that custom setting is true.

Reply:
Done